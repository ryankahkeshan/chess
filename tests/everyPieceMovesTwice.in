game human human
move e2 e4
move d7 d5
move e4 d5
move e7 e6
move d5 e6
move f8 c5
move e6 e7
move e8 e7
move f1 c4
move e7 e8
move c4 b3
move g8 f6
move g1 f3

move e8 g8
move f6 e4
move e1 g1
move e4 c3

move c2 c3

move c2 c4
move f3 e5
move c3 a2
move e5 g6

move g7 g6

move g7 g5
move h7 g6
move h2 h4
move g6 g5
move a1 a2
move g5 g4
move a2 a5

move g4 h3

move e8 f7
move e8 f8
move a5 c5
move a7 a5
move f2 f4

move g4 f4
move g4 f3
move g1 f2
move f3 g2
move f2 e1

move g2 g1 Q

move g2 g1 p

move g2 g1 k

move g2 g1 r

