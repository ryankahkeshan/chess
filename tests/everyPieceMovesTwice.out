8 rnbqkbnr
7 pppppppp
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPPPPPP
1 RNBQKBNR

  abcdefgh

8 rnbqkbnr
7 pppppppp
6  _ _ _ _
5 _ _ _ _ 
4  _ _P_ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqkbnr
7 ppp pppp
6  _ _ _ _
5 _ _p_ _ 
4  _ _P_ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqkbnr
7 ppp pppp
6  _ _ _ _
5 _ _P_ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqkbnr
7 ppp _ppp
6  _ _p_ _
5 _ _P_ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqkbnr
7 ppp _ppp
6  _ _P_ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqk_nr
7 ppp _ppp
6  _ _P_ _
5 _ b _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbqk_nr
7 ppp Pppp
6  _ _ _ _
5 _ b _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbq _nr
7 ppp kppp
6  _ _ _ _
5 _ b _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQKBNR

  abcdefgh

8 rnbq _nr
7 ppp kppp
6  _ _ _ _
5 _ b _ _ 
4  _B_ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh

8 rnbqk_nr
7 ppp _ppp
6  _ _ _ _
5 _ b _ _ 
4  _B_ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh

8 rnbqk_nr
7 ppp _ppp
6  _ _ _ _
5 _ b _ _ 
4  _ _ _ _
3 _B_ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ n _
5 _ b _ _ 
4  _ _ _ _
3 _B_ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ n _
5 _ b _ _ 
4  _ _ _ _
3 _B_ _N_ 
2 PPPP PPP
1 RNBQK _R

  abcdefgh

Invalid move
8 rnbqk_ r
7 ppp _ppp
6  _ _ _ _
5 _ b _ _ 
4  _ _n_ _
3 _B_ _N_ 
2 PPPP PPP
1 RNBQK _R

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ _ _
5 _ b _ _ 
4  _ _n_ _
3 _B_ _N_ 
2 PPPP PPP
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ _ _
5 _ b _ _ 
4  _ _ _ _
3 _Bn _N_ 
2 PPPP PPP
1 RNBQ_RK 

  abcdefgh

Invalid move
Invalid move
8 rnbqk_ r
7 ppp _ppp
6  _ _ _ _
5 _ b N _ 
4  _ _ _ _
3 _Bn _ _ 
2 PPPP PPP
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ _ _
5 _ b N _ 
4  _ _ _ _
3 _B_ _ _ 
2 nPPP PPP
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _ppp
6  _ _ _N_
5 _ b _ _ 
4  _ _ _ _
3 _B_ _ _ 
2 nPPP PPP
1 RNBQ_RK 

  abcdefgh

Invalid move
Invalid move
8 rnbqk_ r
7 ppp _pp 
6  _ _ _p_
5 _ b _ _ 
4  _ _ _ _
3 _B_ _ _ 
2 nPPP PPP
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _pp 
6  _ _ _p_
5 _ b _ _ 
4  _ _ _ P
3 _B_ _ _ 
2 nPPP PP_
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _pp 
6  _ _ _ _
5 _ b _ p 
4  _ _ _ P
3 _B_ _ _ 
2 nPPP PP_
1 RNBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _pp 
6  _ _ _ _
5 _ b _ p 
4  _ _ _ P
3 _B_ _ _ 
2 RPPP PP_
1 _NBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _pp 
6  _ _ _ _
5 _ b _ _ 
4  _ _ _pP
3 _B_ _ _ 
2 RPPP PP_
1 _NBQ_RK 

  abcdefgh

8 rnbqk_ r
7 ppp _pp 
6  _ _ _ _
5 R b _ _ 
4  _ _ _pP
3 _B_ _ _ 
2  PPP PP_
1 _NBQ_RK 

  abcdefgh

Invalid move
Invalid move
8 rnbq k r
7 ppp _pp 
6  _ _ _ _
5 R b _ _ 
4  _ _ _pP
3 _B_ _ _ 
2  PPP PP_
1 _NBQ_RK 

  abcdefgh

8 rnbq k r
7 ppp _pp 
6  _ _ _ _
5 _ R _ _ 
4  _ _ _pP
3 _B_ _ _ 
2  PPP PP_
1 _NBQ_RK 

  abcdefgh

8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _pP
3 _B_ _ _ 
2  PPP PP_
1 _NBQ_RK 

  abcdefgh

8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ PpP
3 _B_ _ _ 
2  PPP _P_
1 _NBQ_RK 

  abcdefgh

Invalid move
8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _ P
3 _B_ _p_ 
2  PPP _P_
1 _NBQ_RK 

  abcdefgh

8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _ P
3 _B_ _p_ 
2  PPP KP_
1 _NBQ_R_ 

  abcdefgh

8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _ P
3 _B_ _ _ 
2  PPP Kp_
1 _NBQ_R_ 

  abcdefgh

8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _ P
3 _B_ _ _ 
2  PPP _p_
1 _NBQKR_ 

  abcdefgh

Invalid move
Invalid move
Invalid move
8 rnbq k r
7 _pp _pp 
6  _ _ _ _
5 p R _ _ 
4  _ _ _ P
3 _B_ _ _ 
2  PPP _ _
1 _NBQKRr 

  abcdefgh

Final Score:
White: 0
Black: 0
